### PR TITLE
ARROW-1662: Move to using Homebrew/bundle and Brewfile

### PR DESCRIPTION
**Pull request by stephengroat**
improves osx dependency management

### CODE REVIEW

File: Brewfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+brew "boost"
+brew "cmake"
+brew "thrift"
+brew "gtk-doc"
+brew "autoconf-archive" 
+brew "gobject-introspection"
+brew "git"
+brew "wget"
+brew "libtool"
+brew "lua"
```
Comment (wesm) [inline Brewfile]: Is it necessary for the Brewfile to be at the top level of the repo?

Some of these dependencies are optional, and also have Category X licenses (https://www.apache.org/legal/resolved.html#category-x). We should not recommend the installation of such components if the are not needed (e.g. for C++ / Python they are not needed).

Comment (stephengroat) [reply Brewfile]: @wesm `Brewfile` can be created at any level, just `cd` in the directory and run `brew bundle`. think that moving the specific dependencies into the specific subdirs is better?

Comment (wesm) [reply Brewfile]: yes, I think that would be better. Thanks

**Comment by stephengroat** [comment]
@wesm updated.

wondering how to get around [license warning](https://travis-ci.org/apache/arrow/jobs/285827256#L1993-L1999)

**Comment by wesm** [comment]
Can brewfiles have comments? If so, then we should add the ASF license header. If not, then add the files to `dev/release/rat_exclude_files.txt`

**Comment by wesm** [comment]
Can you create a JIRA for this and add appropriate prefix to the PR title? see https://github.com/apache/arrow#how-to-contribute

**Comment by stephengroat** [comment]
@wesm got licenses and a jira issue. let me know if i missed anything

**Review by wesm — APPROVED** [review]
+1, looks good, thank you for the contribution!

**Pull request closed without merging**